service/worker/migration/replication.go

```diff
@@ -408,30 +410,43 @@ func (a *activities) checkReplicationOnce(ctx context.Context, waitRequest waitR
 	}
 
 	// check that every shard has caught up
+	readyShardCount := 0
+	logged := false
 	for _, shard := range resp.Shards {
 		clusterInfo, ok := shard.RemoteClusters[waitRequest.RemoteCluster]
-		if !ok {
-			return false, fmt.Errorf("GetReplicationStatus response for shard %d does not contains remote cluster %s", shard.ShardId, waitRequest.RemoteCluster)
-		}
-		if clusterInfo.AckedTaskId == shard.MaxReplicationTaskId {
-			continue // already caught up, continue to check next shard.
+		if ok {
```

Comment:
Do we need to return err if not ok? Do we need to keep retrying if remote cluster can't be found?

Reply:
probably good idea to return error just to surface the issue so we don't need to look at logs to investigate. But should keep retrying. Operator could terminate the workflow if needed.